app/src/main/java/com/zulip/android/networking/AsyncGetEvents.java

```diff
@@ -190,7 +190,13 @@ public Void call() throws Exception {
                             } else {
                                 personDao.update(person);
                             }
-                        } catch (Exception e) {
+                            if (person.getEmail().equals(app.getEmail())) {
```

Comment:
Yes we are calling `setEmail` method, but when I debugged I found that `personCache != null` is false as we are passing `personCache` as null [(here)](https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/models/Person.java#L178) then  as `person` is null [(here)](https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/models/Person.java#L149), cant get person[(here)](https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/models/Person.java#L150) from DB as DB was reset just before `setEmail` call [(here)](https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/activities/ZulipActivity.java#L2117) 
. [At](https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/models/Person.java#L156) this point person is null  now `new Person()` is created with `email, name, avatarURL` where `name` and `avatarURL` are passed as null [(here)](https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/ZulipApp.java#L412). I removed this lines and used [person.equals](https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/models/Person.java#L281) and found that title is again blank. Am I missing out something?